src/scripts/pokemons/PokemonFactory.ts

```diff
@@ -292,4 +298,56 @@ class PokemonFactory {
 
         return null;
     }
+
+    // Gender functions
+    /**
+     * generateGender but using Pokemon ID
+     */
+    public static generateGenderById(id) {
+        const pokemon = PokemonHelper.getPokemonById(id);
+        return this.generateGender(pokemon.gender.ratio, pokemon.gender.type);
+    }
+
+    /**
+     * Calculate which gender has the pokemon.
+     * @param chance Base chance, should be from GameConstants under Gender Ratio comment
+     * @param genderType Gender type (Genderless, male only, etc.), should be from GameConstants under Gender Types comment
+     * @returns {number} 0 = Genderless, 1 = male, 2 = female
+     */
+    public static generateGender(chance: number, genderType: number): number {
```

Comment:
Why is this returning a number and not the enum? If it returned the enum, the rest of the code will be easier to read

Reply:
It was a number before, forgot to change it.